java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java

```diff
@@ -275,6 +275,112 @@ public class PrebuiltSanitizerRules {
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..from.user.displayName")
+                .jsonPath("$..body.content")
+                .jsonPath("$..attachments")
+                .jsonPath("$..mentions[*].mentionText")
+                .jsonPath("$..mentions[*].mentioned.user.displayName")
+                .jsonPath("$..eventDetail.teamDescription")
+                .jsonPath("$..eventDetail.initiator.user.displayName")
+                .build())
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_TEAMS_CHANNELS_MESSAGES_DELTA)
+            .allowedQueryParams(List.of("$select","$top","$skipToken","$expand", "$deltaToken"))
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..from.user.displayName")
+                .jsonPath("$..body.content")
+                .jsonPath("$..attachments")
+                .jsonPath("$..mentions[*].mentionText")
+                .jsonPath("$..mentions[*].mentioned.user.displayName")
+                .jsonPath("$..eventDetail.teamDescription")
+                .jsonPath("$..eventDetail.initiator.user.displayName")
+                .build())
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_USERS_CHATS_MESSAGES)
+            .allowedQueryParams(List.of("$select","$top","$skipToken","$filter", "$orderBy", "$count", "$expand", "$format", "$search", "$skip"))
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..from.user.displayName")
+                .jsonPath("$..body.content")
+                .jsonPath("$..attachments")
+                .jsonPath("$..mentions[*].mentionText")
+                .jsonPath("$..mentions[*].mentioned.user.displayName")
+                .jsonPath("$..eventDetail.teamDescription")
+                .jsonPath("$..eventDetail.initiator.user.displayName")
+                .build())
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_COMMUNICATIONS_CALLS)
+            .allowedQueryParams(List.of("$select","$top","$expand"))
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..displayName")
+                .build())
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_COMMUNICATIONS_CALLRECORDS)
+            .allowedQueryParams(List.of("$select","$expand"))
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..organizer.user.displayName")
+                .jsonPath("$..participants[*].user.displayName")
+                .jsonPath("$..sessions[*].caller.identity.user.displayName")
+                .jsonPath("$..sessions[*].callee.identity.user.displayName")
+                .jsonPath("$..sessions[*].segments[*].caller.identity.user.displayName")
+                .jsonPath("$..sessions[*].segments[*].callee.identity.user.displayName")
```

Comment:
In general, if we could reuse rules the better. For example, all of these rules can be written as:

```suggestion
                .jsonPath("$..user.displayName")
```

`$..` in jsonPath means *any* node; so `$..user.displayName` will be applied to any `user.displayName` node in the json.

Reply:
Done.